src/System.Runtime.Serialization.Formatters/tests/FormatterServicesTests.cs

```diff
@@ -29,21 +31,257 @@ public void GetSerializableMembers_Interface()
+        private interface Interface { }
+
+        public static IEnumerable<object[]> GetUninitializedObject_OpenGenericClass_TestData()
+        {
+            yield return new object[] { typeof(GenericClass<>) };
+            yield return new object[] { typeof(GenericClass<>).MakeGenericType(typeof(GenericClass<>)) };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetUninitializedObject_OpenGenericClass_TestData))]
+        public void GetUninitializedObject_OpenGenericClass_ThrowsMemberAccessException(Type type)
+        {
+            Assert.Throws<MemberAccessException>(() => FormatterServices.GetUninitializedObject(type));
+            Assert.Throws<MemberAccessException>(() => FormatterServices.GetSafeUninitializedObject(type));
+        }
+
+        public interface IGenericClass
+        {
+            int Value { get; set; }
+        }
+
+        public class GenericClass<T> : IGenericClass
+        {
+            public int Value { get; set; }
+        }
+
+        public static IEnumerable<object[]> GetUninitializedObject_ByRefLikeType_TestData()
+        {
+            yield return new object[] { typeof(ArgIterator) };
+            yield return new object[] { typeof(RuntimeArgumentHandle) };
+            yield return new object[] { typeof(TypedReference) };
+        }
+
+        public static IEnumerable<object[]> GetUninitializedObject_ByRefLikeType_NetCore_TestData()
+        {
+            yield return new object[] { typeof(Span<int>) };
+            yield return new object[] { typeof(ReadOnlySpan<int>) };
+            yield return new object[] { Type.GetType("System.ByReference`1[System.Int32]") };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetUninitializedObject_ByRefLikeType_NetCore_TestData))]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.Netcoreapp, "Some runtimes don't support or recognise Span<T>, ReadOnlySpan<T> or ByReference<T> as ref types.")]
+        public void GetUninitializedObject_ByRefLikeType_NetCore_ThrowsNotSupportedException(Type type)
+        {
+            Assert.Throws<NotSupportedException>(() => FormatterServices.GetUninitializedObject(type));
+            Assert.Throws<NotSupportedException>(() => FormatterServices.GetSafeUninitializedObject(type));
+        }
+
+        [Theory]
+        [MemberData(nameof(GetUninitializedObject_ByRefLikeType_TestData))]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "The full .NET framework supports GetUninitializedObject for by ref like types")]
+        public void GetUninitializedObject_ByRefLikeType_NonNetfx_ThrowsNotSupportedException(Type type)
+        {
+            Assert.Throws<NotSupportedException>(() => FormatterServices.GetUninitializedObject(type));
+            Assert.Throws<NotSupportedException>(() => FormatterServices.GetSafeUninitializedObject(type));
+        }
+
+        [Theory]
+        [MemberData(nameof(GetUninitializedObject_ByRefLikeType_TestData))]
```

Comment:
Also, I do not think we need a test in corefx that verifies the buggy netfx behavior (https://github.com/dotnet/corefx/issues/17723#issuecomment-290763235).